codelists/actions.py

```diff
@@ -491,30 +492,84 @@ def export_to_builder(*, version, author):
         CodeObj(version=draft, code=code_obj.code, status=code_obj.status)
         for code_obj in version.code_objs.all()
     )
+    draft, _ = update_draft(draft=draft, source_version=version)
+    return draft
+
+
+def update_draft(*, draft, source_version=None):
+    """
+    Update this draft version in case data have been updated, by:
+    1) recreating the searches, either using the searches from a source version
+       (for a new draft created by `export_to_builder`), or from the searches
+       already on the draft
+    2) checking for any new codes and ensuring the codeset is up to date
+
+    This is called on every GET for a draft, since we can't (currently) tell if the
+    version of a coding system used to create the draft initially is the current one.
+    However, coding systems don't get updated very frequently, so we cache that we've
+    checked it, so we don't have to redo it every time.
+    """
+    cached_updates = cache.get("updated_drafts", [])
+    new_codes = []
+    if draft.pk not in cached_updates:
+        _recreate_draft_searches(source_version or draft, draft)
+        new_codes = _check_new_concepts(draft)
+        cached_updates.append(draft.pk)
+        # cache for 24 hrs
+        cache.set("updated_drafts", cached_updates, timeout=60 * 60 * 25)
```

Comment:
Why `25` and not `24`?

Reply:
typo :)

Reply:
Relief! This caused me to doubt what little I thought I knew about computers 😆